File: database/Document.py
from datetime import date


class Document:
    names = {"registration_num": "Номер регистрации объекта",
             "volume_num": "Номер тома",
             "book_num": "Номер книги",
             "content_quantity": "Количество листов/дисков",
             "inventory_date": "Дата описи",
             "inventory_num": "Номер описи",
             "object_index": "Код/индекс дела",
             "object_name": "Наименование объекта",
             "rack": "Стеллаж №",
             "shelf": "Полка №",
             "expiring_in": "Срок хранения дела",
             "documents_date": "Дата документов",
             "case_num": "Дело № (валовый номер)",
             "destruct_act_num": "Номер акта на уничтожение",
             "destruct_act_date": "Дата акта на уничтожение",
             "struct_division": "Структурное подразделение / отдел",
             "gived_post": "Должность лица, сдавшего в архив",
             "gived_fullname": "ФИО лица, сдавшего в архив",
             "achieved_post": "Должность принявшего со стороны архива",
             "achieved_fullname": "ФИО принявшего со стороны архива",
             "note": "Примечание"}

    def __init__(self,
                 row_id: int,
                 registration_num: str,
                 volume_num: str,
                 book_num: str,
                 content_quantity: int,
                 inventory_date: date,
                 inventory_num: str,
                 object_index: str,
                 object_name: str,
                 rack: int,
                 shelf: int,
                 expiring_in: str,
                 documents_date: date,
                 case_num: str,
                 destruct_act_num: str,
                 destruct_act_date: date,
                 struct_division: str,
                 gived_post: str,
                 gived_fullname: str,
                 achieved_post: str,
                 achieved_fullname: str,
                 note: str):
        self.values = dict()

        self.values["note"] = note
        self.values["achieved_fullname"] = achieved_fullname
        self.values["achieved_post"] = achieved_post
        self.values["gived_fullname"] = gived_fullname
        self.values["gived_post"] = gived_post
        self.values["struct_division"] = struct_division
        self.values["destruct_act_date"] = destruct_act_date
        self.values["destruct_act_num"] = destruct_act_num
        self.values["case_num"] = case_num
        self.values["documents_date"] = documents_date
        self.values["expiring_in"] = expiring_in
        self.values["shelf"] = shelf
        self.values["rack"] = rack
        self.values["object_name"] = object_name
        self.values["object_index"] = object_index
        self.values["inventory_date"] = inventory_date
        self.values["inventory_num"] = inventory_num
        self.values["content_quantity"] = content_quantity
        self.values["book_num"] = book_num
        self.values["volume_num"] = volume_num
        self.values["registration_num"] = registration_num
        self.values["row_id"] = row_id

File: Database/Document.cs
using ArchiveSearchEngine.Database;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ArchiveSearchEngine
{
    public class Document
    {
        public Document(string registrationNum,
            string volumeNum, string bookNum,
            int contentQuantity,
            DateTime inventoryDate, string inventoryNum,
            string objectIndex, string objectName,
            string rack, string shelf, string expiringIn,
            DateTime documentsDate, int caseNum,
            string destructActNum, DateTime destructActDate,
            string structDivision, string givedPost,
            string givedFullname, bool isPersonnel,
            string achievedUsername, string note)
        {
            RegistrationNum = registrationNum;
            VolumeNum = volumeNum;
            BookNum = bookNum;
            ContentQuantity = contentQuantity;
            InventoryDate = inventoryDate;
            InventoryNum = inventoryNum;
            ObjectIndex = objectIndex;
            ObjectName = objectName;
            Rack = rack;
            Shelf = shelf;
            ExpiringIn = expiringIn;
            DocumentsDate = documentsDate;
            CaseNum = caseNum;
            DestructActNum = destructActNum;
            DestructActDate = destructActDate;
            StructDivision = structDivision;
            GivedPost = givedPost;
            GivedFullname = givedFullname;
            IsPersonnel = isPersonnel;
            AchievedUsername = achievedUsername;
            Note = note;
        }
        static public DocumentTable Table { get; set; }

        public string RegistrationNum { get; set; } // 1 Номер регистрации объекта
        public string VolumeNum { get; set; } // 2 Номер тома
        public string BookNum { get; set; } // 3 Номер книги
        public int ContentQuantity { get; set; } // 4 Количество листов / дисков

        public DateTime InventoryDate { get; set; } // 5 Дата описи
        public string InventoryNum { get; set; } // 6 Номер описи

        public string ObjectIndex { get; set; } // 7 Код/индекс дела
        public string ObjectName { get; set; } // 8 Наименование объекта

        public string Rack { get; set; } // 9 Стеллаж
        public string Shelf { get; set; } // 10 Полка

        public string ExpiringIn { get; set; } // 11 Срок хранения дела, по перечню
        public DateTime DocumentsDate { get; set; } // 12 Дата документов
        public int CaseNum { get; set; } // 13 Дело № (валовый номер)

        public string DestructActNum { get; set; } // 14 Номер акта на уничтожение
        public DateTime DestructActDate { get; set; } // 15 Дата акта на уничтожение

        public string StructDivision { get; set; } // 16 Структурное подразделение

        public string GivedPost { get; set; } // 17 Должность сдавшего в архив
        public string GivedFullname { get; set; } // 18 ФИО сдавшего в архив

        public string Note { get; set; } // 27 Примечание

        public bool IsPersonnel { get; set; } // Является ли делом по личному составу

        public bool Available { get { return Table.IsAvailable(RegistrationNum); } } // Доступен ли к выдаче

        public string AchievedUsername { get; } // username того, кто принял документ со стороны архива
    }
}
